src/service/short_url.rs

```diff
@@ -0,0 +1,114 @@
+// Copyright 2024 Zinc Labs Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+use chrono::Utc;
+use config::{get_config, utils::md5};
+use infra::{
+    errors::{DbError, Error},
+    short_url::ShortUrlRecord,
+};
+
+use crate::service::db;
+
+pub fn get_base_url() -> String {
+    let config = get_config();
+    format!("{}{}", config.common.web_url, config.common.base_uri)
+}
+
+/// Shortens the given original URL and stores it in the database
+pub async fn shorten(original_url: &str) -> Result<String, anyhow::Error> {
```

Comment:
As we discussed, here we need to check, if the same hashing `short_url` already exists with different original_url, we need to generate a new `short_url`

Reply:
Handling this case where there is unique violation on `short_id` when doing `db::set`.